tardis/tests/tests_slow/conftest.py

```diff
@@ -87,3 +75,20 @@ def reference(request, reference_datadir):
         luminosity_density_lambda=
             u.Quantity(spectrum['luminosity_density_lambda'], 'erg / (Angstrom s)'))
     return ndarrays
+
+
+@pytest.mark.hookwrapper
+def pytest_runtest_makereport(item, call):
```

Comment:
@yeganer Not exactly, it just appends custom details of our interests (like base64 of image). This is a `hookwrapper` which wraps pytest's own `pytest_runtest_makereport`, and runs before and after the creation (rather, appending details of each method as the tests run).

* `outcome = yield` will yield exactly once (details per test) and return a generator. report object is extracted by `outcome.get_result()`. The name of test method which was tested in a particular iteration is accessed by `report.nodeid` (this is how I am adding an image only to `test_spectrum`).

* Everything before `outcome = yield` is executed first, then `pytest`'s native method executes, which is being generated here by `yield` keyword, and everything after it will be executed afterwards. This happens for every `test_method`.

* This hookwrapper currently works only for `test_spectrum` as `report.nodeid` related if condition evaluates to `True` just once.

* After all the test methods are executed, `pytest-html` does its work by using report object, and essentially the `base64` of image which we had appended as `report.extra` and generating html of the report.

If I was unclear at some place, I'll clarify it further :grinning: